library/core/src/intrinsics.rs

```diff
@@ -2684,18 +2681,15 @@ pub const unsafe fn copy_nonoverlapping<T>(src: *const T, dst: *mut T, count: us
 /// Copies `count * size_of::<T>()` bytes from `src` to `dst`. The source
 /// and destination may overlap.
 ///
-/// If the source and destination will *never* overlap,
-/// [`copy_nonoverlapping`] can be used instead.
-///
-/// `copy` is semantically equivalent to C's [`memmove`], but with the argument
-/// order swapped. Copying takes place as if the bytes were copied from `src`
-/// to a temporary array and then copied from the array to `dst`.
+/// `copy` may temporarily use additional space to copy bytes from `src` into an array, before
```

Comment:
Doesn't it just copy forwards/backwards depending on which region is "lower"? I don't think you ever need a temporary array?

Reply:
arguably a register is in fact a "temporary array", but you're right, I think.